src/TorClient.php

```diff
@@ -0,0 +1,198 @@
+
+        return $this->parseResult($result, PartnerBookingCallResponseBody::class)->getData()->getPartner()->getAllBookings();
+    }
+
+    public function getAccommodation(int $accommodationId): Accommodation
+    {
+        $query = QueryBuilder::createAccommodationQuery($accommodationId);
+
+        $result = $this->runQuery($query);
+
+        return $this->parseResult($result, AccommodationCallResponseBody::class)->getData()->getAccommodation();
+    }
+
+    public function getRentalUnit(int $rentalUnitId): RentalUnit
+    {
+        $query = QueryBuilder::createRentalUnitQuery($rentalUnitId);
+
+        $result = $this->runQuery($query);
+
+        return $this->parseResult($result, RentalUnit::class)->getData()->getRentalUnit();
+    }
+
+    public function createOrReplaceAllotments(int $rentalUnitId, AllotmentCollection $allotmentCollection): AllotmentCollection
+    {
+        $mutation = MutationBuilder::createCreateOrUpdateAllotmentsMutation();
+
+        $variables = ['input' => array_merge(['rentalUnitId' => $rentalUnitId],  $allotmentCollection->toArray())];
+        $result = $this->runQuery($mutation, $variables);
+
+        return $this->parseResult($result, CreateOrReplaceAllotmentsCallResponseBody::class)->getData()->getCreateOrReplaceAllotments();
+    }
+
+    public function createOrReplaceTripPricings(int $rentalUnitId, TripPricingCollection $tripPricingCollection): TripPricingCollection
+    {
+        $mutation = MutationBuilder::createCreateOrUpdateTripPricingsMutation();
+
+        $variables = ['input' => array_merge(['rentalUnitId' => $rentalUnitId], $tripPricingCollection->toArray())];
+        $result = $this->runQuery($mutation, $variables);
+
+        return $this->parseResult($result, CreateOrReplaceTripPricingsCallResponseBody::class)->getData()->getCreateOrReplaceTripPricings();
+    }
+
+    public function deleteTrips(int $rentalUnitId, ?DateTimeInterface $date = null, ?int $duration = null): ?string
+    {
+        $arguments = ['rentalUnitId' => $rentalUnitId];
+        if ($date) {
+            $arguments['date'] = $date->format('Y-m-d');
+        }
+        if ($duration !== null) {
+            $arguments['duration'] = $duration;
+        }
+
+        $mutation = MutationBuilder::createDeleteTripsMutation();
+
+        $variables = ['input' => $arguments];
+        $result = $this->runQuery($mutation, $variables);
+
+        return $this->parseResult($result, DeleteTripsCallResponseBody::class)->getData()->getDeleteTrips()->getMessage();
+    }
+
+    private function runQuery(Query $query, array $variables = []): string
+    {
+        try {
+            $result = $this->client->runQuery($query, false, $variables);
+            if (!$result->getResponseBody()) {
+                throw new BadResponseException('No response body found');
+            }
+            return $result->getResponseBody();
+        } catch (\Exception $exception) {
+            throw $exception;
```

Comment:
Dit slaat volgens mij nergens op...

Reply:
Ik zie ook niet helemaal wat er nu gebeurt met "nette" graphql errors, die zouden in elk geval incl. message etc. terug te vinden moeten zijn via de Client lijkt me.